curator-framework/src/main/java/org/apache/curator/framework/imps/CreateBuilderImpl.java

```diff
@@ -485,7 +485,6 @@ private String protectedPathInForeground(String adjustedPath, byte[] data) throw
         }
         catch ( Exception e)
         {
-            ThreadUtils.checkInterrupted(e);
```

Comment:
As a correction to the previous comment, this fixes the issue by exposing the interruption using the interrupt status only, and not throwing an exception, rather than doing both as it is at the moment.